Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -88,58 +81,64 @@ public class CRTClientEngine: HttpClientEngine {
-        let crtRequest = request.toHttpRequest(bufferSize: windowSize)
-        let streamReader: StreamReader = DataStreamReader()
+        let crtRequest = try request.toHttpRequest()
+        var responseData = Data()
+        
+        let makeStatusCode: (HTTPStream) -> HttpStatusCode = { stream in
+            guard
+                let statusCodeInt = try? stream.statusCode(),
+                let statusCode = HttpStatusCode(rawValue: statusCodeInt)
+            else { return .notFound }
+            return statusCode
+        }
 
-        let requestOptions = HttpRequestOptions(request: crtRequest) { [self] (stream, _, httpHeaders) in
+        let requestOptions = HTTPRequestOptions(request: crtRequest) { [self] (stream, _, httpHeaders) in
             logger.debug("headers were received")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode)) ?? HttpStatusCode.notFound
+            response.statusCode = makeStatusCode(stream)
             response.headers.addAll(httpHeaders: httpHeaders)
         } onIncomingHeadersBlockDone: { [self] (stream, _) in
             logger.debug("header block is done")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode)) ?? HttpStatusCode.notFound
+            response.statusCode = makeStatusCode(stream)
         } onIncomingBody: { [self] (stream, data) in
             logger.debug("incoming data")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode)) ?? HttpStatusCode.notFound
-            let byteBuffer = ByteBuffer(data: data)
-            streamReader.write(buffer: byteBuffer)
+            response.statusCode = makeStatusCode(stream)
+            responseData.append(data)
         } onStreamComplete: { [self] (stream, error) in
             logger.debug("stream completed")
-            streamReader.hasFinishedWriting = true
-            if case let CRTError.crtError(unwrappedError) = error {
-                if unwrappedError.errorCode != 0 {
-                    logger.error("Response encountered an error: \(error)")
-                    streamReader.onError(error: ClientError.crtError(error))
-                    continuation.resume(throwing: error)
-                    return
-                }
+            if let error = error, error.code != 0 {
+                logger.error("Response encountered an error: \(error)")
+                continuation.resume(throwing: CommonRunTimeError.crtError(error))
+                return
             }
-
-            response.body = .stream(.reader(streamReader))
-
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode)) ?? HttpStatusCode.notFound
+        
+            response.body = .stream(.buffer(ByteBuffer(data: responseData)))
+            response.statusCode = makeStatusCode(stream)
 
             continuation.resume(returning: response)
         }
         return requestOptions
     }
+    
+    public func close() async {
+        // no-op
```

Comment:
There no longer is anything to close on the CRT side of things and so we just no-op here. 
We can remove the `close` function from the interface as well, and then delete it here but I left it in as it might sense to "close" other http engine implementations.

Reply:
would suggest deleting this entirely. We can do before go GA but not later.